src/main/java/pika/task/Task.java

```diff
@@ -0,0 +1,71 @@
+package pika.task;
+
+/**
+ * Base Task class
+ */
+public class Task { //Base Task Class
+    protected boolean isDone;
+    private final String name;
+
+    /**
+     * Constructor of Task class
+     *
+     * @param name Name of Task
+     */
+    public Task(String name) {
+        this.name = name;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as done when called and return it
+     */
+    public String done() {
```

Comment:
Naming it markAsDone or completeTask might be more descriptive?

Reply:
Yes thanks for pointing this out!